### ISSUE TITLE
Use cluster number in column 2 on H-lines in uc files

### ISSUE THREAD

**Issue by claudiuskerth** [torognes/vsearch#207]
Hi,

I have trouble with the uclust output from `vsearch --cluster_fast`. The second column should contain the cluster number (ID). In my run of `vsearch` it had found 240,511 clusters. Accordingly, the highest ID in lines containing cluster records (C) is 240,510. The highest number in lines for cluster centroids (S) is also 240,510. However, the cluster id numbers for matching sequences (H) are much higher. The highest number - 9,176,669 - is just a little less than the total number of input sequences.

I hope that's a quick fix :-)

Below is the vsearch log from the run:

> vsearch v2.3.0_linux_x86_64, 94.4GB RAM, 24 cores
> vsearch --cluster_fast totaluniqseq.SE.fasta.gz --consout /dev/fd/63 --clusterout_id --clusterout_sort --id 0.8 --qmask dust --centroids /dev/fd/62 --uc /dev/fd/61 --log totaluniqseq.SE.log 
> Started  Wed Oct 19 17:12:42 2016422257966 nt in 9179521 seqs, min 46, max 46, avg 46

```
  Alphabet  nt
Word width  8
 Word ones  8
    Spaced  No
    Hashed  No
     Coded  No
   Stepped  No
     Slots  65536 (65.5k)
   DBAccel  100%
```

> Clusters: 240511 Size min 1, max 152808, avg 38.2
> Singletons: 56552, 0.6% of seqs, 23.5% of clusters
> 
> Finished Wed Oct 19 17:21:58 2016
> Elapsed time 09:16
> Max memory 2.9GB

**Comment by torognes** [comment]
Hi, you are right, the contents of the second column on the H lines is not as documented. Instead of the cluster number it actually contains the ordinal number of the matching input sequence. If it matches the first sequence it would be 0, if it matches the second sequence it would be 1, and so on. To find the cluster it belongs to you need to look at the last column (10). This is similar to what usearch does (but not to its documentation). 

I'd like to keep the compatibility with usearch. We need to update the documentation.

Thank you for reporting this issue.

**Comment by frederic-mahe** [comment]
Hi, I pushed a new unit test to cover that point. I will update the documentation soon.

**Comment by claudiuskerth** [comment]
Hi guys,

thank you for the high-speed reaction.

Just to clarify, "ordinal number of the _matching_ input sequence" means "ordinal number of the centroid sequence that the hit sequence (H) matches". I am assuming "ordinal number" means that every input sequence is assigned a number. The first input sequence gets the number 0, the second input sequence gets the number 1 and so on.

I can write a parser to get the cluster ID's for hit sequences (H), but I would find a uclust-like output that reports the cluster ID for each input sequence very useful. So may I suggest such an output as enhancement?

**Comment by torognes** [comment]
To your first point: yes, that is correct.

We'll consider adding your suggestion as a feature in a future release.

**Comment by claudiuskerth** [comment]
Thanks Torbjorn and thank you for an already awesome `Vsearch`! It would be hard if not impossible to do my analyses without it.

**Comment by torognes** [comment]
I was wrong. This is a bug and I have misled you. Sorry!

Usearch does report the cluster number in the second column on the H-lines in the UC files.

I will fix the bug and release a new vsearch version very soon.

**Comment by torognes** [comment]
Resolved in version 2.3.2 just released. Documentation updated.
